src/vm/callingconvention.h

```diff
@@ -1255,7 +1270,17 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
         if (thValueType.IsHFA())
         {
             CorElementType type = thValueType.GetHFAType();
-            cFPRegs = (type == ELEMENT_TYPE_R4)? (argSize/sizeof(float)): (argSize/sizeof(double));
+            bool isSinglePrecision = (type == ELEMENT_TYPE_R4);
+
+            cFPRegs = isSinglePrecision ? (argSize/sizeof(float)): (argSize/sizeof(double));
+            m_hasArgLocDescForStructInRegs = true;
+
+            m_argLocDescForStructInRegs.Init();
+            m_argLocDescForStructInRegs.m_cGenReg = 0;
+            m_argLocDescForStructInRegs.m_cFloatReg = cFPRegs;
+            m_argLocDescForStructInRegs.m_idxGenReg = m_idxGenReg;
+            m_argLocDescForStructInRegs.m_idxFloatReg = m_idxFPReg;
+            m_argLocDescForStructInRegs.m_isSinglePrecision = isSinglePrecision;
```

Comment:
I understand that you have modelled your changes as done for unix_amd64. I don't really like the way this has been implemented for amd64_unix. I dont think above needs to be done here in GetNextOffset(). GetNextOffset() only returns the offset in TransitionBlock. In order to get ArgLocDesc we have another function GetArgLoc(). So first you get the offset and then use that to get ArgLocDesc. This would also eliminate the need for fields m_hasArgLocDescForStructInRegs & m_argLocDescForStructInRegs in ArgIterator which I think is not required.